app/src/full/kotlin/io/homeassistant/companion/android/sensors/LocationSensorManager.kt

```diff
@@ -928,18 +928,24 @@ class LocationSensorManager :
         if (updateLocationAs == SEND_LOCATION_AS_ZONE_ONLY) {
             val zones = getZones(serverId)
             val locationZone = zones
-                .filter { !(it.attributes["passive"] as Boolean) && it.containsWithAccuracy(location) }
-                .minByOrNull { (it.attributes["radius"] as Number).toFloat() }
+                .filter {
+                    val passive = it.attributes["passive"] as? Boolean
+                    val radius = it.attributes["radius"] as? Number
+                    return@filter passive == false && it.containsWithAccuracy(location) && radius != null
```

Comment:
```suggestion
                    return@filter passive == false && radius != null && it.containsWithAccuracy(location)
```

One final tweak as it appears you didn't check `containsWithAccuracy` - which [uses the radius attribute](https://github.com/home-assistant/android/blob/4b9f001feb456383b825095fb1d76a6e128b93d2/common/src/main/kotlin/io/homeassistant/companion/android/common/data/integration/ZoneAttributes.kt#L17)